### PR TITLE
Feature/us8158 trip scholarship tool

### PR DESCRIPTION
**Pull request by silbermm**

### CODE REVIEW

File: crds-mp-tools/Views/Shared/SelectionError.cshtml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿@{
```
Comment (andycanterbury) [inline crds-mp-tools/Views/Shared/SelectionError.cshtml]: There's nothing in here that is selection specific, can we replace the Shared/Error.cshtml with this one? This one is reusable and doesn't have the text that the other was which we wouldn't really want to release into prod anyway.

File: crds-mp-tools/Views/TripScholarship/TripScholarship.cshtml [hunk 2]
```diff
@@ -1,21 +1,55 @@
...
+            <div class="form-group  @Directives.InputClass(Html.ViewData, "SelectedTrip")">
+                @Html.LabelFor(model => model.Trips, new { @class = "form-control-label" })
+                @Html.DropDownListFor(model => model.SelectedTrip, Model.Trips, "-- Please Select One --", new { @class = "form-control" })
+                @Html.ValidationMessageFor(model => model.SelectedTrip, null, new { @class = "form-control-feedback" })
+            </div>  
+            
+            <div class="form-group @Directives.InputClass(Html.ViewData, "ScholarshipAmount")">
+                @Html.LabelFor(model => model.ScholarshipAmount, new { @class = "form-control-label" })
+                <div class="input-group">
+                    <span class="input-group-addon">$</span>
```
Comment (andycanterbury) [inline crds-mp-tools/Views/TripScholarship/TripScholarship.cshtml]: I hadn't seen bootstrap's input-group-addon before. Slick. 👍

File: CI/SQL/20170621_102100_US8158-ConfigurationSettingsForTripScholarships.sql [hunk 3]
```diff
@@ -0,0 +1,59 @@
+USE MinistryPlatform;
+Go
+
+DECLARE @APP_CODE nvarchar(50) = 'CRDS-COMMON';
```
Comment (silbermm) [inline CI/SQL/20170621_102100_US8158-ConfigurationSettingsForTripScholarships.sql]: what else would you suggest? I need to have a type, so I just used the one that is used for that field.

File: MinistryPlatform/Models/MpDonationDistribution.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+namespace MinistryPlatform.Models
+{
+    [MpRestApiTable(Name = "Donation_Distributions")]
+    public class MpDonationDistribution
+    {
+        [JsonProperty(PropertyName = "Donation_Distribution_ID")]
+        public int DonationDistributionId { get; set; }
+
+        [JsonProperty(PropertyName = "Donation_ID")]
+        public int DonationId { get; set; }
```
Comment (silbermm) [inline MinistryPlatform/Models/MpDonationDistribution.cs]: we don't

File: crds-mp-tools-common/Selections/SelectionService.cs [hunk 5]
```diff
@@ -9,25 +9,44 @@ namespace crds_mp_tools_common.Selections
...
             try
             {
+                if (toolParams.RecordId != null && toolParams.RecordId > 0)
+                {
+                    return new MpSelection
+                    {
+                        // ReSharper disable once PossibleInvalidOperationException
+                        PageId = (int) toolParams.PageId,
+                        SelectionCount = 1,
+                        UserId = userId,
```
Comment (silbermm) [inline crds-mp-tools-common/Selections/SelectionService.cs]: yes, this is the userId that we pass in via the currently logged in user

Comment (silbermm) [inline crds-mp-tools-common/Selections/SelectionService.cs]: yes. we pass in the userid to the method

File: crds-mp-tools-common/Selections/TooManySelectionsException.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace crds_mp_tools_common.Selections
+{
+    public class TooManySelectionsException : Exception
```
Comment (silbermm) [inline crds-mp-tools-common/Selections/TooManySelectionsException.cs]: I think the type is enough to understand what's happening, don't you?

File: crds-mp-tools/Controllers/TripScholarshipController.cs [hunk 7]
```diff
@@ -8,27 +13,73 @@ namespace crds_mp_tools.Controllers
...
+                    {
+                        TempData?.Add("alert", "Please close this tool reopen after selecting only 1 donor.");
+                        return View("SelectionError");
+                    }
+                    TempData?.Add("alert", $"There was an error: {ex.Message}");
+                    return View("Error");
+                });            
+        }
+
+        [Authorize]
```
Comment (silbermm) [inline crds-mp-tools/Controllers/TripScholarshipController.cs]: Yes, but that wasn't available when I did this work

Comment (andycanterbury) [inline crds-mp-tools/Controllers/TripScholarshipController.cs]: So you aren't going to do it?

**Pull request merged**